src/blocks/toggleSwtich/ToggleSwitch.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+
+import styled from 'styled-components';
+import * as Switch from '@radix-ui/react-switch';
+
+import { Box } from '../box';
+import { Text } from '../text';
+
+export type ToggleSwitchProps = {
+  /* Label for the toggle switch */
+  label?: string;
+  /* Description for the toggle switch */
+  description?: string;
+  /* Sets toggle switch as disabled */
+  disabled?: boolean;
+  /* Render the toggle before text contents */
+  leadingToggle?: boolean;
+  /* Sets toggle switch to checked */
+  checked: boolean;
+  /* Function invoked when toggle switch is clicked*/
+  onCheckedChange: (checked: boolean) => void;
+};
+
+const StyledToggleRoot = styled(Switch.Root)`
+  width: 38px;
+  height: 20px;
+  padding: 3px;
+  background-color: var(--components-toggle-switch-background-unselected, #c4cbd5);
+  border-radius: var(--radius-md);
+  position: relative;
+  &[data-state='checked'] {
+    background-color: var(--components-toggle-switch-background-selected, #c742dd);
+  }
+  &:disabled {
+    cursor: not-allowed;
+    background: var(--components-toggle-switch-background-disabled, #eaebf2);
+    span {
+      background: var(--components-toggle-switch-icon-disabled, #b0b3b9);
+    }
+  }
+`;
+const StyledToggleThumb = styled(Switch.Thumb)`
+  display: block;
+  cursor: pointer;
+  width: 14px;
+  height: 14px;
+  background-color: var(--components-toggle-switch-icon-default, #fff);
+  border-radius: var(--radius-round);
+  transition: transform 100ms;
+  will-change: transform;
+  &[data-state='checked'] {
+    transform: translateX(17.5px);
+  }
+`;
+const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
+  label,
+  description,
+  disabled = false,
+  onCheckedChange,
+  leadingToggle = true,
+  checked,
+}) => {
+  return (
+    <Box
```

Comment:
Also let's use styled components instead of box in the design system components. Reason if tomorrow Box is broken or deprecated then our other components works as expected. Every design system component should be standalone and not dependent on any other outside thing.

Reply:
okay , got it , but should we also replace text component too ?

Reply:
okay doing it